perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ServicesManagerBlImpl.java

```diff
@@ -379,6 +377,9 @@ private ServiceAttributes getData(PerunSession sess, Service service, Resource r
 	}
 
 	private ServiceAttributes getData(PerunSession sess, Service service, Facility facility, Resource resource, boolean filterExpiredMembers) {
+		service.setUseExpiredMembers(!filterExpiredMembers);
+		servicesManagerImpl.updateService(sess, service);
```

Comment:
I would maybe compare `service.isUseExpiredMembers()` and `filterExpiredMembers` and update the service only if they differ (so it should be updated at most once). We should probably call updateService in the BL layer instead of impl layer because an audit message is created there. What do you think?

Reply:
Thank you for your suggestion, BL version of update will be much better. Also, service will be updated only if corresponding value is changed, however, I was thinking whether:

```**!filterExpiredMembers != service.isUseExpiredMembers()**```

would not be more clear, but now, there is simplified version of it.